reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportDefinitionIT.kt

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+package com.amazon.opendistroforelasticsearch.reportsscheduler.rest
+
+import com.amazon.opendistroforelasticsearch.reportsscheduler.PluginRestTestCase
+import com.amazon.opendistroforelasticsearch.reportsscheduler.ReportsSchedulerPlugin.Companion.BASE_REPORTS_URI
+import com.amazon.opendistroforelasticsearch.reportsscheduler.validateErrorResponse
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestStatus
+import org.junit.Assert
+
+class ReportDefinitionIT : PluginRestTestCase() {
+    private fun constructReportDefinitionRequest(
+        trigger: String = """
+            "trigger":{
+                "triggerType":"OnDemand"
+            },
+        """.trimIndent(),
+        name: String = "report_definition"
+    ): String {
+        return """
+            {
+                "reportDefinition":{
+                    "name":"$name",
+                    "isEnabled":true,
+                    "source":{
+                        "description":"description",
+                        "type":"Dashboard",
+                        "origin":"localhost:5601",
+                        "id":"id"
+                    },
+                    "format":{
+                        "duration":"PT1H",
+                        "fileFormat":"Pdf",
+                        "limit":1000,
+                        "header":"optional header",
+                        "footer":"optional footer"
+                    },
+                    $trigger
+                    "delivery":{
+                        "recipients":["nobody@email.com"],
+                        "deliveryFormat":"LinkOnly",
+                        "title":"title",
+                        "textDescription":"textDescription",
+                        "htmlDescription":"optional htmlDescription",
+                        "channelIds":["optional_channelIds"]
+                    }
+                }
+            }
+            """.trimIndent()
+    }
+
+    fun `test create, get, update, delete report definition`() {
+        var reportDefinitionRequest = constructReportDefinitionRequest()
+        var reportDefinitionResponse = executeRequest(
```

Comment:
Should we make these `val` vs. `var` so that they are immutable?

Reply:
I reused those variables for other requests, won't work if they are immutable

Reply:
I think it is better to use val and use another variable if they are not related.

Reply:
will refactor, thanks